src/sentry/auth/access.py

```diff
@@ -76,6 +79,20 @@ def _sso_params(member: OrganizationMember) -> Tuple[bool, bool]:
     return requires_sso, sso_is_valid
 
 
+@dataclass(frozen=True, eq=True)
+class TeamMembership:
```

Comment:
what do we think about this alternative:  put `has_scope` on the omt object, and for sentry apps, could we create a new `Access` child class that simply returns `true` for `has_team_access`?

Reply:
(we would pass in omts instead of teams)

Reply:
I like this.

The new `Access` subclass for Sentry apps is raising some questions for me about the meaning of the `teams` and `projects` attributes for cases where `has_*_access` is overridden. It kind of breaks the model for `teams` to be an empty set when `has_team_access` conveys the meaning of "belongs to all teams". The unit tests produced some complaints along those lines when I tried to refactor. (Preview: rather than a new subclass, I'd like if `_from_sentry_app` returned an `OrganizationGlobalAccess` instance.)

I have, I think, a pretty good idea about having `teams` and `projects` be overridable properties, rather than always simple data attributes. It's more complex, but worth it IMHO to keep the little white lies out of the model.

Reply:
https://github.com/getsentry/sentry/pull/31880 😮‍💨